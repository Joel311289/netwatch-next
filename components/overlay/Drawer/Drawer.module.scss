.wrapper {
  &.left {
    border-top-right-radius: var(--border-radius-portal);
    border-bottom-right-radius: var(--border-radius-portal);
  }

  &.right {
    border-top-left-radius: var(--border-radius-portal);
    border-bottom-left-radius: var(--border-radius-portal);
  }
}

.content {
  padding: 30px var(--padding-aside-overlay);
  width: 360px;
  max-width: 100vw;
}
