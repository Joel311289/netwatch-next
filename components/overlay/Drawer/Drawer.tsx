import DrawerMUI from '@mui/material/Drawer';
import { ModalProps } from '@mui/material';
import { ElementChildren, ElementHTML, FunctionVoid } from '@/types';
import { useModal } from '@/hooks/useModal';
import { Closable } from '@/components/layout';
import { PaperModalProps } from '@/components/overlay/Portal/Portal';
import { classes } from '@/utils/helpers';
import styles from '@/components/overlay/Drawer/Drawer.module.scss';

export type DrawerProps = Partial<typeof defaultProps> &
  ElementHTML &
  ElementChildren & {
    heading?: string;
    opened?: boolean;
    onChange?: FunctionVoid<boolean>;
    position?: 'left' | 'right';
    modalConfig?: PaperModalProps;
  };

const defaultProps = {
  position: 'right'
};

const Drawer = ({
  className,
  children,
  position,
  heading,
  opened,
  onChange,
  modalConfig
}: DrawerProps) => {
  const { isOpened, handleChange } = useModal(opened, onChange);

  return (
    <DrawerMUI
      aria-labelledby="drawer"
      classes={{
        paper: classes(styles.wrapper, position && styles[position], className)
      }}
      ModalProps={(modalConfig || {}) as Partial<ModalProps>}
      anchor={position}
      open={isOpened}
      onClose={() => handleChange(false)}>
      <Closable heading={heading} onClose={() => handleChange(false)}>
        <div className={styles.content}>{children}</div>
      </Closable>
    </DrawerMUI>
  );
};

Drawer.defaultProps = defaultProps;

export default Drawer;
