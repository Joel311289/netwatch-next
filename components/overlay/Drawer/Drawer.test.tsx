import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Drawer from '@/components/overlay/Drawer/Drawer';

const portalHeading = 'Panel';
const portalContent = 'Content';

const content = () => <div>{portalContent}</div>;

describe('Tests Drawer component', () => {
  test('renders component correctly', () => {
    const { container } = render(<Drawer opened>Test</Drawer>);

    expect(container).toMatchSnapshot();
  });

  test('renders component correctly', () => {
    render(
      <Drawer opened heading={portalHeading}>
        {content()}
      </Drawer>
    );

    expect(screen.getByText(portalHeading)).toBeInTheDocument();
    expect(screen.getByText(portalContent)).toBeInTheDocument();
  });

  test('closes portal correctly', async () => {
    render(
      <Drawer opened heading={portalHeading}>
        {content()}
      </Drawer>
    );

    expect(screen.getByText(portalContent)).toBeInTheDocument();

    userEvent.click(screen.getByLabelText('close'));

    await waitFor(() => {
      expect(screen.queryByText(portalContent)).not.toBeInTheDocument();
    });
  });
});
