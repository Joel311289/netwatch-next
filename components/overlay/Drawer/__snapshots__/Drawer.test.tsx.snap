// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Drawer component renders component correctly 1`] = `
<div
  aria-hidden="true"
/>
`;
